### DISCUSSION TITLE
Extend the Solidus API for Solidus extensions?

### DISCUSSION THREAD

**Discussion by morismael** [solidusio/solidus#4386] [Troubleshooting]
Hello. Working on a marketplace using solidus_papyal_marketplace gem which doesn't have an API. Does the Solidus API provide a way to extend it to extensions or should I develop my own API for this extensions? Thanks.

**Comment by jarednorman** [comment]
Extensions can provide API controllers. Is that what you're asking?

**Reply by morismael** [reply]
There is no API module attached to the Paypal marketplace extension. My question is how can I extend it to have an API? Maybe via the existing base Solidus API.

**Reply by jarednorman** [reply]
Just like with any other kind of controller you want to add, you can add an API controller and add routes for it:

```ruby
class Spree::Api::MyCustomPaypalController < Spree::Api::BaseController
  # your custom actions
end
```

**Reply by morismael** [reply]
Thanks, let me try that.

**Reply by kennyadsl** [reply]
@morismael can you please mark this answer as "the answer" if this solved this issue?

**Reply by morismael** [reply]
Hello. Sorry for the delay. Eventually I found another approach and no longer using the Paypal marketplace gem. So I don't know if I should mark this as an answer as I haven't tested. Please advise.

**Reply by kennyadsl** [reply]
It's ok, thanks anyway for letting us know!
